QuantLib/ql/models/model.hpp

```diff
@@ -139,8 +139,18 @@ namespace QuantLib {
         virtual boost::shared_ptr<Lattice> tree(const TimeGrid&) const = 0;
     };
 
+
     // inline definitions
 
+
+    inline Real AffineModel::discountBondOption(Option::Type type,
+                                                Real strike,
+                                                Time maturity,
+                                                Time , // not used... ?!?!
```

Comment:
In the default implementation the bond start is assumed to be equal to the option maturity (i.e. zero settlement delay), you can overwrite the function with a more sophisticated version taking into account the delay (this is for example done in the Hull White Model). The JamshidianSwaptionEngine (for example) will provide more accurate pricings of market swaptions then.